Integrations/MISP_V2/MISP_V2.py

```diff
@@ -128,6 +125,71 @@ def warn(*args):
 ''' HELPER FUNCTIONS '''
 
 
+def filter_obj(obj: dict, is_att: bool = True) -> dict:
+    """ Filter Event, Attribute
+
+    Args:
+        obj(dict): dictionary MISP event or attribute.
+        is_att: True if dict represent Attribute otherwise Event (False)
+
+    Returns:
+        dict: Filtered dict by configuration.
+    """
+    metadata_state = demisto.params().get('metadata')
+    related_events_state = demisto.params().get('related_events')
+    selected_keys: List[str] = [item.lower() for item in demisto.params().get('context_select')]
+    ignored_keys_metadata: List[str] = ["Galaxy", "Tag"]
+    ignored_keys_related_events: List[str] = ["RelatedEvent"]
+
+    for dict_key in list(obj.keys()):
+        if dict_key in ignored_keys_metadata:
+            if not metadata_state:
+                obj[dict_key] = []
+            else:
+                continue
+        elif dict_key in ignored_keys_related_events:
+            if not related_events_state:
+                obj[dict_key] = []
+            else:
+                continue
+        elif is_att and selected_keys and dict_key not in selected_keys:
+            obj.pop(dict_key)
```

Comment:
why in the rest you are clearing the data but here you are completely removing the key?

Reply:
Because it influences other parts of the code, I wanted to pop them all but creating ec or readable it will cause an error due to not iterable object.